src/praisonai-agents/praisonaiagents/memory/adapters/factories.py

```
@@ -101,6 +101,47 @@ def create_mongodb_memory_adapter(**kwargs) -> MemoryProtocol:
+        MemoryProtocol adapter instance.
+
+    Raises:
+        ImportError: If the ``dakera`` SDK is not installed.
+    """
+    try:
+        import dakera  # noqa: F401
+    except ImportError:
+        raise ImportError(
+            "dakera is required for the dakera adapter. "
+            "Install with: pip install 'praisonaiagents[dakera]' (or: pip install dakera)"
+        )
```

Ruff (0.15.20)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)